classes/Language.php

```diff
@@ -1335,37 +1336,57 @@ public static function updateMultilangFromClass($table, $className, $lang)
             }
         }
     }
-    
+
     /**
      * Language::installRtlStylesheets()
-     * @param bool $bo_theme
-     * @param bool $fo_theme
-     * @param null $theme_name
-     * @param null $iso
+     *
+     * @param bool $boTheme
+     * @param bool $foTheme
+     * @param string|null $themeName
+     * @param string|null $iso
      * @param bool $install
-     * @param null $path
+     * @param string|null $path
+     *
+     * @throws \PrestaShop\PrestaShop\Core\Localization\RTL\Exception\GenerationException
+     * @throws Exception
      */
-    public static function installRtlStylesheets($bo_theme = false, $fo_theme = false, $theme_name = null, $iso = null, $install = false, $path = null)
+    public static function installRtlStylesheets($boTheme = false, $foTheme = false, $themeName = null, $iso = null, $install = false, $path = null)
     {
-        $admin_dir = ($install) ? _PS_ROOT_DIR_.'/admin/' : _PS_ADMIN_DIR_.'/';
-        $front_dir = _PS_ROOT_DIR_.'/themes/';
         if ($iso) {
-            $lang_pack = Language::getLangDetails($iso);
+            $lang_pack = static::getLangDetails($iso);
             if (!$lang_pack['is_rtl']) {
                 return;
             }
         }
-        
-        if ($bo_theme) {
-            \RTLGenerator::generate($admin_dir.'themes');
+
+        $generator = new StylesheetGenerator();
+
+        // generate stylesheets for BO themes
+        if ($boTheme) {
+            if (defined('_PS_ADMIN_DIR_')) {
+                $adminDir = _PS_ADMIN_DIR_;
+            } else {
+                $adminDir = _PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'admin';
+                $adminDir = (is_dir($adminDir)) ? $adminDir : ($adminDir.'-dev');
+                $adminDir .= DIRECTORY_SEPARATOR;
```

Comment:
I can guess this is used during the installation of PrestaShop.
Why don't you use the `$install` param anymore ?

Reply:
This is done during install (if you choose an RTL language during install) and also whenever you add an RTL language package.

`$install` is useless now really, but I decided to leave it there in case it's needed in the future.